ballerina/service_types.bal

```diff
@@ -14,18 +14,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-# Triggers when Choreo recieves a new message from Azure service bus. Available action: onMessage
-public type MessageService service object {
-    # Triggers when a new message is received from Azure service bus
-    # + message - The Azure service bus message recieved
-    # + caller - The Azure service bus caller instance
-    # + return - Error on failure else nil()
-    isolated remote function onMessage(Message message, Caller caller) returns error?;
+# The ASB service type.
+public type Service distinct service object {
+    // isolated remote function onMessage(asb:Message message, asb:Caller caller) returns error?;
 
-    # Triggers when there is an error in message processing
-    #
-    # + context - Error message details  
-    # + error - Ballerina error
-    # + return - Error on failure else nil()
-    isolated remote function onError(ErrorContext context, error 'error) returns error?;
+    // isolated remote function onError(asb:MessageRetrievalError 'error) returns error?;
```

Comment:
Any reason to comment out the `onMessage()` and  `onError()` contracts from the service object?

Reply:
This is to offer flexibility to developer. `onError` method is optional and hence if we define it in the contract we won't be able to achieve it. And the parameters for `onMessage` method is can change (it can have the `asb:Caller` or only have `asb:Message`). Hence, we comment out there from the contract and future planning to introduce a proper compiler plugin validation for this.

Reply:
got it. +1 to do the validations via a compiler plugin as a future improvement. And we better create an issue for tracking purposes.

Reply:
Ack. Captured here [1]

[1] - https://github.com/ballerina-platform/ballerina-library/issues/6549